### PR TITLE
finishes ticket 4

### PR DESCRIPTION
**Pull request by ellamarym**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -32,3 +32,31 @@ describe('GET /api/topics' , () => {
     });
 })
 
+describe('GET /api/articles', () => {
+  test('responds with an array of objects, with author, title, article-id, topic, created-at, votes, and comment count keys', () => {
+    return request(app)
+    .get('/api/articles')
+    .expect(200)
+    .then(({body}) => {
+      body.articles.forEach((article) => {
```
Comment (callumu21) [inline __tests__/app.test.js:42]: When using a .forEach loop in tests, just make sure you also check the array length. If the returned array were empty, this test would still pass as the .forEach would be skipped.

File: models/app.model.js [hunk 2]
```diff
@@ -1,9 +1,28 @@
...
+exports.fetchArticles = () => {
+    return db.query(`
+    SELECT title, topic, articles.author, articles.article_id, articles.created_at, articles.votes, COUNT(comments.article_id) AS comment_count
+    FROM articles
+    LEFT JOIN comments ON comments.article_id = articles.article_id
+    GROUP BY articles.article_id
+    ORDER BY articles.created_at DESC
+    ;
+    `).then((articles)=> {
+       const countToNums = articles.rows.map((article) => {
```
Comment (callumu21) [inline models/app.model.js]: It looks as though this logic is to coerce the string value for comment_count to a number, which should ideally be done as part of the SQL query for better efficiency. I would look at using CAST() or double colon notation (::) for changing it automatically to a number.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Hey, Ella. Just a few changes to be made before approval. Please see my comments, and if you're unsure of anything just pop me a message.

**Review by JVal-NC — COMMENTED** [review]
All good, go ahead and merge. Please just bear in mind git comments regarding branching.

**Pull request merged**